docs/features/tuples.md

```diff
@@ -0,0 +1,227 @@
+
+A lot of details and motivation for the feature is given in 
+[https://github.com/dotnet/roslyn/issues/347](https://github.com/dotnet/roslyn/issues/347)
+
+Current design notes:
+https://github.com/dotnet/roslyn/issues/10429
+
+NOTE: The goal of this document is capture what is being implemented. As design evolves, the document will undergo adjustments. 
+Changes in design will be applied to this document as the changes are implemented.
+
+-----------------------------------
+
+Tuple types
+-----------
+
+Tuple types would be introduced with syntax very similar to a parameter list:
+
+```C#
+public (int sum, int count) Tally(IEnumerable<int> values) { ... }
+    
+var t = Tally(myValues);
+Console.WriteLine($"Sum: {t.sum}, count: {t.count}");   
+ ```
+ 
+The syntax `(int sum, int count)` indicates an anonymous struct type with public fields of the given names and types.
+
+With no further syntax additions to C#, tuple values could be created as
+``` c#
+var t1 = new (int sum, int count) (0, 1);
+var t2 = new (int sum, int count) { sum = 0, count = 0 };
+```
+
+Tuple literals
+--------------
+
+``` c#
+var t1 = (0, 2); 				// infer tuple type from values
+var t2 = (sum: 0, count: 1);	// infer tuple type from names and values
+```
+
+Creating a tuple value of a known target type:
+``` c#
+public (int sum, int count) Tally(IEnumerable<int> values) 
+{
+    var s = 0; var c = 0;
+    foreach (var value in values) { s += value; c++; }
+    return (s, c); // target typed to (int sum, int count)
+}
+```
+
+Note that specifying field names is optional, however when names are provided, all fields must be named. Duplicate names are disallowed.
+
+
+``` c#
+var t1 = (sum: 0, 1);		// error! some fields are named some are not.
+var t2 = (sum: 0, sum:1);	// error! duplicate names.
+```
+
+
+Duality with underlying type
+--------------
+
+Tuples map to underlying types of particular names - 
+
+```
+System.ValueTuple<T1, T2>
+System.ValueTuple<T1, T2, T3>
+...
+System.ValueTuple<T1, T2, T3,..., T7, TRest>
+```
+
+In all scenarios tuple types behave exactly like underlying types with only additional optional enhancement of the more expressive field names given by the programmer.
```

Comment:
Yes, this needs to be changed. There are scenarios (like field access) where the tuple and underlying type are observably different. In some cases this can affect lambda inference and overload resolution.